deeplearning4j-nn/src/main/java/org/deeplearning4j/datasets/iterator/parallel/MultiBoolean.java

```diff
@@ -0,0 +1,93 @@
+package org.deeplearning4j.datasets.iterator.parallel;
+
+import lombok.extern.slf4j.Slf4j;
+import org.nd4j.linalg.exception.ND4JIllegalStateException;
+
+/**
+ * This is utility class, that allows easy handling of multiple joint boolean states
+ *
+ * @author raver119@gmail.com
+ */
+@Slf4j
+public class MultiBoolean {
```

Comment:
Probably yes. And probably i should just make it package-private. The only use for this thing is supposed to be single-threaded & for consumers state tracking. So, limited basically to 8 independent states.